scripts/util.py

```diff
@@ -301,66 +301,83 @@ def subscription_list(project_id=None, page_size=None, slurm_cluster_name=None):
     return subscriptions
 
 
-def subscription_create(subscription_id, project_id=None):
-    """Create pub/sub subscription"""
+def delete_subscriptions(seq):
+    """Deletes a list of subscriptions
+    API calls are made in parallel from thread pool
+    """
     from google.cloud import pubsub_v1
-    from google.api_core import exceptions
 
-    if project_id is None:
-        project_id = lkp.project
-    topic_id = lkp.cfg.pubsub_topic_id
-
-    publisher = pubsub_v1.PublisherClient()
     subscriber = pubsub_v1.SubscriberClient()
-    topic_path = publisher.topic_path(project_id, topic_id)
-    subscription_path = subscriber.subscription_path(project_id, subscription_id)
 
     with subscriber:
-        request = {
-            "name": subscription_path,
-            "topic": topic_path,
-            "ack_deadline_seconds": 60,
-            "labels": {
-                "slurm_cluster_name": cfg.slurm_cluster_name,
-            },
-        }
-        try:
-            subscription = subscriber.create_subscription(request=request)
-            log.info(f"Subscription created: {subscription_path}")
-            log_subscriptions.debug(f"{subscription}")
-        except exceptions.AlreadyExists:
-            log.info(f"Subscription '{subscription_path}' already exists!")
+        with ThreadPoolExecutor() as exe:
+            futures = []
+            for i in seq:
+                future = exe.submit(_delete_subscription, i, subscriber)
+                futures.append(future)
+            for future in as_completed(futures):
+                result = future.exception()
+                if result is not None:
+                    raise result
 
 
-def subscription_delete(subscription_id, project_id=None):
+def _delete_subscription(subscription_id, subscriber):
     """Delete pub/sub subscription"""
-    from google.cloud import pubsub_v1
     from google.api_core import exceptions
 
-    if project_id is None:
-        project_id = lkp.project
+    project_id = lkp.project
+    subscription_path = subscriber.subscription_path(project_id, subscription_id)
+    try:
+        subscriber.delete_subscription(request={"subscription": subscription_path})
+        log.info(f"Subscription deleted: {subscription_path}.")
+    except exceptions.NotFound:
+        log.info(f"Subscription '{subscription_path}' not found!")
 
+
+def create_subscriptions(seq):
```

Comment:
Could we use a more expressive variable name / type hint here?

Reply:
Done